projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts

```diff
@@ -0,0 +1,287 @@
+
+    const address = new AddressModel();
+    address.addressLine1 = 'Address Line 1';
+    address.addressLine2 = 'Address Line 2';
+    address.addressLine3 = 'Address Line 3';
+    address.postTown = 'postTown';
+    address.county = 'county';
+    address.postCode = 'postCode';
+    address.country = 'country';
+
+    wrapperComponent.addressField = new FormControl(address);
+    fixture.detectChanges();
+
+    expect(debugElement.query($POSTCODE_LOOKUP)).toBeTruthy();
+    expect(debugElement.query($SELECT_ADDRESS)).toBeFalsy();
+
+    /* expect(debugElement.query($ADDRESS_COMPLEX_FIELD)).toBeTruthy();
+    expect(debugElement.query($ADDRESS_COMPLEX_FIELD).nativeElement['hidden']).toBeFalsy();
+
+    expect(testHostComponent.componentUnderTest.writeComplexFieldComponent.complexGroup.value).toEqual(address); */
+
+  });
+
+  it('should render a single option of \'No address found\' when no addresses are returned from AddressService', () => {
+
+    spyOn(addressService, 'getAddressesForPostcode').and.returnValue(of([]));
+
+    queryPostcode(POSTCODE);
+
+    expect(debugElement.query($MANUAL_LINK)).toBeTruthy();
+    expect(addressService.getAddressesForPostcode).toHaveBeenCalledWith('P05TCDE');
+    expect(debugElement.query($SELECT_ADDRESS)).toBeTruthy();
+    expect(debugElement.query($ADDRESS_LIST).children.length).toEqual(1);
+    expect(debugElement.query($ADDRESS_LIST).children[0].nativeElement.innerHTML.trim()).toEqual('No address found');
+
+  });
+
+  it('should render a default \'summary item\' and 3 address options when 3 addresses are returned from AddressesService', () => {
+
+    const address2 = buildAddress(2);
+    address2.addressLine2 = '';
+    const address3 = buildAddress(3);
+    address3.addressLine3 = '';
+
+    spyOn(addressService, 'getAddressesForPostcode').and.returnValue(
+      of([buildAddress(1), address2, address3])
+    );
+
+    queryPostcode(POSTCODE);
+
+    expect(addressService.getAddressesForPostcode).toHaveBeenCalledWith('P05TCDE');
+    expect(debugElement.query($MANUAL_LINK)).toBeTruthy();
+    expect(debugElement.query($SELECT_ADDRESS)).toBeTruthy();
+    expect(debugElement.query($ADDRESS_LIST).children.length).toEqual(4);
+    expect(debugElement.query($ADDRESS_LIST).children[0].nativeElement.innerHTML.trim()).toEqual('3 addresses found');
+
+    expect(debugElement.query($ADDRESS_LIST).children[1].nativeElement.innerHTML.trim()).toEqual(
+      'addressLine1-1, addressLine2-1, addressLine3-1, postTown-1'
+    );
+    expect(debugElement.query($ADDRESS_LIST).children[2].nativeElement.innerHTML.trim()).toEqual(
+      'addressLine1-2, addressLine3-2, postTown-2'
+    );
+    expect(debugElement.query($ADDRESS_LIST).children[3].nativeElement.innerHTML.trim()).toEqual(
+      'addressLine1-3, addressLine2-3, postTown-3'
+    );
+
+  });
+
+  /* it('should populate the address with the option selected, removing the \'manual link\'', () => {
```

Comment:
@johnbenjamin-hmcts - as above... should these tests be reinstated?

Reply:
@connorpgpmcelroy - could you please check and revert?

Reply:
@Daniel-Lam - I have updated the code.